src/vs/platform/issue/electron-main/issueMainService.ts

```diff
@@ -442,6 +442,26 @@ export class IssueMainService implements IIssueMainService {
 		});
 	}
 
+	async $sendReporterStatus(extensionId: string, extensionName: string): Promise<boolean[]> {
+		const window = this.issueReporterWindowCheck();
+		const replyChannel = `vscode:sendReporterStatus`;
+		return Promises.withAsyncBody<boolean[]>(async (resolve) => {
+			const cts = new CancellationTokenSource();
+			const result = [false, false];
+			window.sendWhenReady(replyChannel, cts.token, { replyChannel, extensionId, extensionName });
+			validatedIpcMain.on('vscode:receiveReporterStatus', (_: unknown, data: boolean[]) => {
+				resolve(data);
+			});
+			try {
+				await timeout(2000);
```

Comment:
I know you're just copying me, but a nice pattern is something like:
https://github.com/microsoft/vscode/blob/f49bc367f1efe0e1dde9d85b2fdea5e65c683b5d/src/vs/base/common/async.ts#L114

```ts
const result = await raceTimeout(
  new Promise(resolve => validatedIpcMain.once('vscode:receiveReporterStatus', (_: unknown, data: boolean[]) => resolve(data)),
  2000,
  () => cts.cancel()
);

resolve(result ?? defaultResult)
```

it handles the timeout for you and cancels it when your promise resolves. There may be a nicer way to wrap the event in a promise, I only did a small search.

Not blocking with this comment.

Reply:
changed it up for `getReporterStatus` and it works nicely, barring the `as boolean[]`. Not sure if there's a better way around that? Debt is to clean up the rest of the functions as well.